oslo_vmware/rw_handles.py

```diff
@@ -721,13 +723,26 @@ def read(self, chunk_size):
     def tell(self):
         return self._bytes_read
 
+    def _shutdown_connection(self):
+        """Gracefully shuts down the socket of the HTTP(S) connection"""
+        # HTTPSConnection wraps the TCP socket into an SSL transport,
+        # which needs to be shutdown first.
+        if 'WrappedSocket' in type(self._conn.sock).__name__:
+            _socket = self._conn.sock.socket
+            self._conn.sock.shutdown()
```

Comment:
Trying to find out how the interface of `shutdown()` looks like here, I cannot find a `WrappedSocket` object (I've greped through my local Python installation for that string, Python 3.8). Afaics, it should be an `SSLSocket` object [`wrap_socket()` docs](https://docs.python.org/3/library/ssl.html#ssl.SSLContext.wrap_socket), which seems to have the same interface for `shutdown()` and as I understand it should call the socket's `shutdown()` [docs](https://docs.python.org/3/library/ssl.html#ssl.SSLSocket)

Can you help me understand this, please?

Reply:
thank you.